File: infra/src/main/java/com/beat/infra/persistence/member/repository/MemberJpaRepository.java
package com.beat.infra.persistence.member.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.beat.domain.member.domain.SocialType;
import com.beat.infra.persistence.member.entity.MemberJpaEntity;

public interface MemberJpaRepository extends JpaRepository<MemberJpaEntity, Long> {
	@Query("SELECT m FROM Member m WHERE m.socialId = :socialId AND m.socialType = :socialType")
	Optional<MemberJpaEntity> findBySocialTypeAndSocialId(
		@Param("socialId") Long socialId,
		@Param("socialType") SocialType socialType
	);
}

File: core/infra/src/main/kotlin/com/beat/infra/persistence/member/repository/MemberJpaRepository.kt
package com.beat.infra.persistence.member.repository

import com.beat.domain.member.model.SocialType
import com.beat.infra.persistence.member.entity.MemberJpaEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.Optional

interface MemberJpaRepository : JpaRepository<MemberJpaEntity, Long> {
    @Query("SELECT m FROM Member m WHERE m.socialId = :socialId AND m.socialType = :socialType")
    fun findBySocialTypeAndSocialId(
        @Param("socialId") socialId: Long?,
        @Param("socialType") socialType: SocialType,
    ): Optional<MemberJpaEntity>
}
